packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +290,102 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed']
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, nameserver_ip)
+      query_dns = Resolv::DNS.new(nameserver: [nameserver_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      serial = soa_record.serial if soa_record
+      {
+        ip: ip,
+        serial: serial
+      }
+    end
+
+    def new_dns_values
+      new_vals = OpenStruct.new
+
+      new_vals.dns_server = @scenario_answers['foreman_proxy']['dns_server']
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.reverse_zone = [@scenario_answers['foreman_proxy']['dns_reverse']].flatten.first
```

Comment:
Reverse zones in the installer can be an array. If you're updating SOA records, shouldn't you update them all in a loop? Why is the first one special?

Reply:
@ekohl If it's possible to have multiple reverse zones legitimately, then it makes sense to loop over them.  But I don't understand how or why one would ever have multiple reverse zones for the satellite server specified in the answer file, since you only have one record for the satellite server and a reverse zone is for reverse IP lookups.  My assumption was that someone just formatted the yaml file that way and it stuck.

Reply:
No, this is a legitimate feature. People are actually annoyed that we don't have a parameter for multiple forward zones so the Katello manual even documents a way to achieve this (https://theforeman.org/plugins/katello/3.14/installation/index.html#multiple-subnets-and-domains). Your script will not catch these, but parsing `/etc/foreman-installer/custom-hiera.yaml` is probably a bad idea.

>  since you only have one record for the satellite server and a reverse zone is for reverse IP lookups

This is a misunderstanding of the feature. If you provision new hosts, foreman-proxy can (if it manages the zone) create reverse DNS entries. If you have multiple subnets, you probably want to manage them all.

> My assumption was that someone just formatted the yaml file that way and it stuck.

Its data type is `Optional[Variant[String, Array[String]]]` (see https://github.com/theforeman/puppet-foreman_proxy/blob/ef0d8745eecccd6475a9b8d75b917c3eaf01b45b/manifests/init.pp#L428)

Reply:
@ekohl Thanks for clarifying.  Will update to loop over the array.